api/src/models/user.rs

```diff
@@ -29,6 +32,31 @@ impl User {
     ) -> dsl::Filter<users::table, WithUsername<'_>> {
         users::table.filter(Self::with_username(username))
     }
+
+    /// Build and execute query to grant each of the given roles to the user.
+    pub fn add_roles_x(
+        &self,
+        conn: &PgConnection,
+        roles: &[RoleType],
+    ) -> Result<usize, diesel::result::Error> {
+        // Map the role types to strings. No need to collect this iter, just
+        // pass it along directly
+        let role_names = roles.iter().map(|role_type| role_type.to_str());
+        diesel::insert_into(user_roles::table)
+            .values(
+                roles::table
+                    .select((
+                        self.id.into_sql::<sql_types::Uuid>(),
+                        roles::columns::id,
+                    ))
+                    .filter(roles::columns::name.eq_any(role_names)),
+            )
+            .into_columns((
+                user_roles::columns::user_id,
+                user_roles::columns::role_id,
+            ))
+            .execute(conn)
+    }
```

Comment:
Another thing that I'm not using in this PR, but I'll use in the next one for sure.